parts/k8s/kubernetescustomscript.sh

```diff
@@ -496,6 +496,25 @@ configAddons() {
     fi
 }
 
+installAzCli(){
+    AZ_REPO=$(lsb_release -cs)
+    echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ $AZ_REPO main" | \
+    sudo tee /etc/apt/sources.list.d/azure-cli.list
+    retrycmd_if_failure 3 1 30 curl -L https://packages.microsoft.com/keys/microsoft.asc | sudo apt-key add -
+    apt_get_update && apt_get_install 2 20 120 azure-cli
+}
+
+exitScript() {
+    if [[ "$SAS_TOKEN" != "" ]] && [[ "$CONTAINER_NAME" != "" ]] && [[ "$STORAGE_ACCOUNT_NAME" != "" ]] && [[ "$BLOB_PREFIX" != "" ]]; then
+        installAzCli || echo "Failed to install az cli"
+        set +x
+        az storage blob upload -f /var/log/azure/cluster-provision.log -c $CONTAINER_NAME -n $BLOB_PREFIX-`hostname` --account-name $STORAGE_ACCOUNT_NAME --sas-token $SAS_TOKEN || echo "Failed to upload to Storage Account"
```

Comment:
I guess if the installAzCli fails above we'll just hit the or condition here on file not found?